### PR TITLE
feat: add `secondary` variant to links and the introduce a secondary core button

### PR DESCRIPTION
**Pull request by sebald**
# Description

Added `secondary` variants to the `<Link>` component so it can be effectively used as a secondary "button". Also had to add a `secondary` button to the core ... because there is none.

- [x] This change requires a UI-Kit update - inform Alex Tirado (@tirado-rx)

# What should be tested?

Does a link and button with `variant=secondary` look identical :D 

## Are they some special informations required to test something?

# Reviewers:

@marigold-ui/developer
@marigold-ui/designer

### CODE REVIEW

File: themes/theme-b2b/src/components/Link.styles.ts [hunk 1]
```diff
@@ -6,6 +6,10 @@ export const Link: ThemeComponent<'Link'> = cva(
     variants: {
       variant: {
         menuItemLink: 'text-text-base hover:no-underline',
+        secondary: [
```
Comment (sarahgm) [inline themes/theme-b2b/src/components/Link.styles.ts:9]: can you add this to storybook please?

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: themes/theme-b2b/src/components/Link.styles.ts (hunk 1 above)
Comment (sebald) [reply themes/theme-b2b/src/components/Link.styles.ts:9]: done

Comment (aromko) [inline themes/theme-b2b/src/components/Link.styles.ts:9]: There is a test in packages/components/src/Link/Link.test.tsx which could be adjusted. And I think there is a variant used which isn't actual any more?

Comment (sebald) [reply themes/theme-b2b/src/components/Link.styles.ts:9]: If you are referring to the "second" the theme is just a fixture for testing. We don't have any real themes in the tests.

**Review by sarahgm — APPROVED** [review]

**Pull request merged**